app/src/main/java/br/com/catbag/gifreduxsample/models/AppState.java

```diff
@@ -23,4 +32,40 @@ public boolean getHasMoreGifs() {
         return true;
     }
 
-}
+    public String toJson() throws JsonProcessingException {
+        return new ObjectMapper().writeValueAsString(this);
+    }
+
+    public static AppState fromJson(String json) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new GuavaModule());
+        return mapper.readValue(json, AppState.class);
+    }
```

Comment:
I would recommends to write simple unit tests for those methods, only to ensures the right behavior.